wpilib/wpilib/drive/vector2d.py

```diff
@@ -0,0 +1,47 @@
+# validated: 2017-10-23 TW 9dc1de1d09da edu/wpi/first/wpilibj/drive/Vector2d.java
+import math
+
+
+class Vector2d:
+    """This is a 2D vector struct that supports basic operations"""
+    def __init__(self, x=0.0, y=0.0):
+        """Construct a 2D vector
+
+        :param x: x component of the vector
+        :param y: y component of the vector
+        """
+        self.x = x
+        self.y = y
+
+    def rotate(self, angle):
+        """Rotate a vector in Cartesian space.
+
+        :param angle: Angle in degrees by which to rotate vector counter-clockwise
+        """
+        angle = math.radians(angle)
+        cosA = math.cos(angle)
+        sinA = math.sin(angle)
+
+        self.x = self.x * cosA - self.y * sinA
+        self.y = self.x * sinA + self.y * cosA
```

Comment:
This is wrong, very wrong, and doesn't match upstream. How did we not catch this?

Reply:
I would say it's subtly wrong from upstream, as only the last line is incorrect. Good catch.